From: Beryl 7 reproducible build <beryl7-build@localhost>
Subject: [PATCH] tailscale: backport 1.102.3 from openwrt-25.12

Backport the package recipe and service hardening from OpenWrt packages
commit 5d6ecf4803ed041c9f76271e1d1e27e976a68c72 without moving the other
pinned package feeds.  Client self-update is deliberately disabled because
the binary is managed by apk / the firmware image.

---
 net/tailscale/Makefile             | 6 +++---
 net/tailscale/files/tailscale.init | 5 +++++
 2 files changed, 8 insertions(+), 3 deletions(-)

diff --git a/net/tailscale/Makefile b/net/tailscale/Makefile
index b4f4ac4..93994c1 100644
--- a/net/tailscale/Makefile
+++ b/net/tailscale/Makefile
@@ -7,12 +7,12 @@
 include $(TOPDIR)/rules.mk
 
 PKG_NAME:=tailscale
-PKG_VERSION:=1.94.1
+PKG_VERSION:=1.102.3
 PKG_RELEASE:=1
 
 PKG_SOURCE:=$(PKG_NAME)-$(PKG_VERSION).tar.gz
 PKG_SOURCE_URL:=https://codeload.github.com/tailscale/tailscale/tar.gz/v$(PKG_VERSION)?
-PKG_HASH:=80eec367cabd6012a668233b11ca5c6df29b0dd24c7cfd74b71958c93ec9b644
+PKG_HASH:=0e94d961c31ce7d33e8b7ce4ac6fdbec83ee5658784eed69eb7fce300729d717
 
 PKG_MAINTAINER:=Zephyr Lykos <self@mochaa.ws>, \
 		Sandro Jäckel <sandro.jaeckel@gmail.com>
@@ -26,6 +26,6 @@ GO_PKG:=tailscale.com/cmd/tailscaled
 GO_PKG_LDFLAGS:=-X 'tailscale.com/version.longStamp=$(PKG_VERSION)-$(PKG_RELEASE) (OpenWrt)'
 GO_PKG_LDFLAGS_X:=tailscale.com/version.shortStamp=$(PKG_VERSION)
-GO_PKG_TAGS:=ts_include_cli,ts_omit_aws,ts_omit_bird,ts_omit_completion,ts_omit_kube,ts_omit_systray,ts_omit_taildrop,ts_omit_tap,ts_omit_tpm
+GO_PKG_TAGS:=ts_include_cli,ts_omit_aws,ts_omit_bird,ts_omit_clientupdate,ts_omit_completion,ts_omit_kube,ts_omit_systray,ts_omit_taildrop,ts_omit_tap,ts_omit_tpm
 
 include $(INCLUDE_DIR)/package.mk
 include ../../lang/golang/golang-package.mk
diff --git a/net/tailscale/files/tailscale.init b/net/tailscale/files/tailscale.init
index 57f36c5..d215864 100755
--- a/net/tailscale/files/tailscale.init
+++ b/net/tailscale/files/tailscale.init
@@ -6,5 +6,7 @@ USE_PROCD=1
 START=80
 
+export TS_NO_LOGS_NO_SUPPORT=true
+
 start_service() {
   local state_file
   local port
@@ -26,6 +28,9 @@ start_service() {
   # Use nftables by default - can be changed to 'iptables' in tailscale config
   procd_set_param env TS_DEBUG_FIREWALL_MODE="$fw_mode"
 
+  # Disable logging to log.tailscale.com
+  procd_append_param env TS_NO_LOGS_NO_SUPPORT=true
+
   # Set the port to listen on for incoming VPN packets.
   # Remote nodes will automatically be informed about the new port number,
   # but you might want to configure this in order to set external firewall
-- 
2.47.3
